pkg/test/framework/components/gcemetadata/gce_metadata_server.yaml

```diff
@@ -0,0 +1,46 @@
+# Copyright Istio Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+apiVersion: v1
+kind: Service
+metadata:
+  name: gce-metadata-server
+  labels:
+    app: gce-metadata
+spec:
+  ports:
+  - name: http
+    port: 8080
+  selector:
+    app: gce-metadata
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: gce-metadata-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: gce-metadata
+  template:
+    metadata:
+      labels:
+        app: gce-metadata
+    spec:
+      containers:
+      - image: gcr.io/istio-testing/fake-gce-metadata:1.0
```

Comment:
Does the fame metadata server require a filepath or env to response with information? or is it all baked in?

Reply:
At some point, it can be expanded to take envs / filepaths for additional config, but at the moment, it is all baked in.

My tentative plan for this is to get all the extremely basic pieces working and in, and then refine/expand as needed to cover more and more use cases. The framework is fairly complicated, so I'm hoping just to get an end-to-end working before getting fancy.